new-entrypoint.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -e
+
+function is_puma_installed() {
+	[ ! -f Gemfile.lock ] && return 1
+	grep ' puma ' Gemfile.lock >/dev/null
+}
+
+function check_number() {
+	if [[ ! "$2" =~ ^[0-9]+$ ]]; then
+		echo "$1 needs to be a non-negative number"
+		exit 1
+	fi
+}
+
+if [ "$APPLICATION_TYPE" = "compliance-backend" ]; then
+	check_number PUMA_WORKERS     "${PUMA_WORKERS:-0}"
+	check_number PUMA_MIN_THREADS "${PUMA_MIN_THREADS:-0}"
+	check_number PUMA_MAX_THREADS "${PUMA_MAX_THREADS:-0}"
```

Comment:
This syntax means the default is `0` right? That doesn't seem appropriate :thinking: Maybe 1 for all of them?

Reply:
I grabbed this from [here](https://github.com/RedHatInsights/s2i-insights-compliance/blob/82d848378312103a3e1836cc788049acdf91820a/s2i/bin/run) , which at the same time looks "inspired" by [this guy](https://github.com/sclorg/s2i-ruby-container/blob/master/2.7/s2i/bin/run#L16-L18)

> This syntax means the default is `0` right?

yup. From `man bash`:
```
${parameter:-word}
    Use Default Values.  If parameter is unset or null, the expansion of word is substituted.  Otherwise, the value of parameter is substituted.
```
I don't know what effect would be to have 0.

Reply:
IMO it's okay to accept a 0 , regardless if you find that unlikely or useless - it's a valid value after all - I'd keep it as it is for consistency with the existing reference scripts

Reply:
I've confirmed that trying to run puma with `0` for these three values works fine and actually uses 5 threads (the default) and 1 worker.

Reply:
Awesome, I guess then this actually makes sense, which is not a surprise.